### PR TITLE
Suggestions mode

### PR DESCRIPTION
**Pull request by fjrdomingues**
Adds a mode where an agent (advisor) creates a written suggestion for the task instead of a code patch.

### CODE REVIEW

File: ui.js [hunk 1]
```diff
@@ -125,28 +162,23 @@ async function main(task, test=false) {
...
+  let solutions
 
-    if (autoApply){
-      // This actually applies the solution to the file
-      updateFile(file.path, coderRes);
-    }
+  // Depending on Suggestion Mode call different agents
+  if (!suggestionMode) {
+    // Ask agent for code changes (coder agent)
+    solutions = await getSolution(relevantFiles, autoApply)
```
Comment (ProfessorX737) [inline ui.js:178]: The arguments here don't match the function definition of getSolution :/ 

That's why typescript is so much better

File: ui.js [hunk 2]
```diff
@@ -104,13 +104,50 @@ async function getTask(task, options){
...
+  }
+  const files = getFiles(relevantFiles)
+  if (files.length == 0) throw new Error("No relevant files found")
+  return files
+}
+
+async function getSolution(task, files, autoApply) {
+  let solutions = [];
+  for (const file of files) {
+    const coderRes = await runAgent(agents.coder, task, file, interactive);
```
Comment (ProfessorX737) [inline ui.js:122]: Need to pass in `interactive`

File: modules/summaries.js [hunk 3]
```diff
@@ -59,7 +59,7 @@ function getFiles(files){
   let retFiles=[]
   for (const file of files) {
     const pathToFile = file.path;
-    const fileContent = fs.readFileSync(pathToFile, 'utf8');
+    const fileContent = fs.readFileSync(path.posix.join(process.env.CODE_DIR, pathToFile), 'utf8');
```
Comment (ProfessorX737) [inline modules/summaries.js:62]: Could you also make the same update in `fsOutput.js:56`

**Comment by fjrdomingues** [comment]
Thanks @ProfessorX737 I'll fix this once we have some other huge PR being merged. Will probably have to fix conflicts

**Pull request merged**